onnx/onnx-data.in.proto

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) ONNX Project Contributors.
+// Licensed under the MIT license.
+
+syntax = "proto2";
+
+package {PACKAGE_NAME};
+// #if ONNX-ML
+import "onnx/onnx-ml.proto";
+// #else
+import "onnx/onnx.proto";
+// #endif
+
+// This file contains the proto definitions for MapProto and
+// SequenceProto. These protos are used to represent the data structures
+// of maps and sequence for use in test data or ModelProto.
+
+// Sequences
+//
+// Defines a dense, ordered, collection of elements that are of homogeneous types.
+// Sequences can be made out of tensors, maps, or sequences.
+message SequenceProto {
+
+  optional string name = 1;
+
+  enum DataType {
+    UNDEFINED = 0;
+    TENSOR = 1;
+    SPARSE_TENSOR = 2;
+    SEQUENCE = 3;
+    MAP = 4;
+  }
+
+  // The data type of the element.
+  // This field MUST have a valid SequenceProto.DataType value
+  optional int32 elem_type = 2;
+
+  // For TensorProto values.
+  // When this field is present, the elem_type field MUST be TENSOR.
+  repeated TensorProto tensor_values = 3;
+
+  // For SparseTensorProto values.
+  // When this field is present, the elem_type field MUST be SPARSE_TENSOR.
+  repeated SparseTensorProto sparse_tensor_values = 4;
+
+  // For SequenceProto values, allowing sequences to be of themselves.
+  // When this field is present, the elem_type field MUST be SEQUENCE.
+  repeated SequenceProto sequence_values = 5;
+
+  // For MapProto values.
+  // When this field is present, the elem_type field MUST be MAP.
+  repeated MapProto map_values = 6;
+
+}
+
+
+// Maps
+//
+// Specifies an associative table, defined by a key type and a value type.
+message MapProto {
+
+  optional string name = 1;
+
+  // All MapProto data types must have the same length of keys and values.
+
+  // The data type of the key.
+  // This field MUST have a valid TensorProto.DataType value of
+  // INT8, INT16, INT32, INT64, UINT8, UINT16, UINT32, UINT64, or STRING
+  optional int32 key_type = 2;
+
+  // Every element of keys has to be one of the following data types
+  // INT8, INT16, INT32, INT64, UINT8, UINT16, UINT32, UINT64, or STRING.
+  // The integer cases are represented by the repeated int64 field keys below.
+  repeated int64 keys = 3;
+
+  // If keys are strings, they are represented by the repeated bytes field
+  // string_keys below.
+  repeated bytes string_keys = 4;
+
+  // The data type of the value.
+  // This field MUST have a valid SequenceProto.DataType value of
+  // TENSOR, SPARSE_TENSOR, MAP, SEQUENCE
+  optional int32 value_type = 5;
```

Comment:
this is not needed, right? The type has been specified in SequenceProto already.

Reply:
Removed! Thanks for the feedback.